attestation.go

```diff
@@ -39,14 +38,13 @@ type AttestationHashes struct {
 	appKeyHash [sha256.Size]byte // Sometimes set, depending on application.
 }
 
-// Serialize returns a byte slice that contains our concatenated hashes.  Note
-// that all hashes are always present.  If a hash was not initialized, it's set
-// to 0-bytes.
+// Serialize returns a byte slice that contains our concatenated hashes.
+// hashPrefix defines the hash type and length.  Note that all hashes are
+// always present.  If a hash was not initialized, it's set to 0-bytes.
 func (a *AttestationHashes) Serialize() []byte {
-	str := fmt.Sprintf("%s%s%s%s%s",
+	str := fmt.Sprintf("%s%s%s%s",
```

Comment:
i'm unsure if `fmt.Sprintf` is a robust fit for creating a binary value. it may be better to use `append` or something similar

Reply:
There's also a go-multihash package we could use to perform the encoding at insert time, so we only need to concatenate the two slices. That might be cleaner overall.

Reply:
> There's also a [go-multihash](https://pkg.go.dev/github.com/multiformats/go-multihash) package we could use to perform the encoding at insert time

Yeah, i suppose you could, but I think what you wrote above looks fine & lightweight enough. You could probably drop the `[:]` and just keep the `...` at the end of the slice that is to be appended. A shortened version could look something like this:
```
serialized := []byte{}
serialized = append(serialized, append(hashPrefix, a.tlsKeyHash...)...)
serialized = append(serialized, append(hashPrefix, a.appKeyHash...)...)
return serialized
```

Reply:
or even:

```
serialized := append(hashPrefix, a.tlsKeyHash...)
serialized = append(serialized, append(hashPrefix, a.appKeyHash...)...)
return serialized
```

Reply:
The `[:]` is required because the AttestationHashes members are arrays, not slices.

Reply:
good point, edited suggestion above

Reply:
I actually like your first suggestion better, since it aligns the two calls and makes it more obvious what's different between the lines. See f04c24d.